scanner/indexer/indexer.go

```diff
@@ -311,7 +363,18 @@ func (i *localIndexer) IndexContainerImage(
 			Headers: layerReq.Header,
 		})
 	}
-	return i.libIndex.Index(ctx, manifest)
+
+	ir, err := i.libIndex.Index(ctx, manifest)
+	if err != nil {
+		return nil, err
+	}
+
+	err = i.metadataStore.StoreManifest(ctx, manifestDigest.String(), randomExpiry(time.Now()))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
> I figure if we return a failure, we'll retry at some point, and it'll eventually succeed. What do you think?

You're right this may trigger immediate retries in the current scanner v4 client (and we reprocessing eventually after that).  However the retry will trigger a `GetIndexReport` first, which looks like will successfully pull the index report even if the manifest expiry storage failed. 

By returning the index report even if the storage fails you save a round trip.  If `GetIndexReport` is modified to require the manifest expiry exists prior to returning any index report then I can see it making more sense to error as is or self correct - ie: on retrieval if the index report is found but the manifest expiry is not, then insert an expiry (self correct).

Reply:
I addressed it in https://github.com/stackrox/stackrox/pull/12648/commits/3ae55f519f5b0ce541d8ade274f48202be61b9fa